### ISSUE TITLE
2.3.0-beta.1: subscribing to audio_levels causes crashes when destroying SDK while in conference

### ISSUE THREAD

**Issue by kubaau** [DolbyIO/comms-sdk-cpp#4]
With this piece of code:
```
#include <dolbyio/comms/sdk.h>
#include <cstdio>

int main() {
  auto cppsdk = dolbyio::comms::sdk::create(token, [](auto&&) {});

  cppsdk->conference()
      .add_event_handler([](const dolbyio::comms::audio_levels&) {})
      .on_error([](auto&&) {});

  cppsdk->session()
      .open({})
      .then([sdk = cppsdk.get()](auto&&) {
        return sdk->conference().demo(
            dolbyio::comms::spatial_audio_style::disabled);
      })
      .on_error([](auto&&) {});

  while (getchar() != 'q') {
  };
}
```
The application will crash on exit instead of exiting cleanly.

**Comment by kubaau** [comment]
Also, on Windows including the sdk.h causes the following warning:
```
media_engine.h(115,51): warning C4305: 'initializing': truncation from 'double' to 'const float'
```
See:
```
static constexpr float speaking_threshold = 0.05;
```

**Comment by djova-dolby** [comment]
It appears this crash is only happening on Windows, I was unable to reproduce on MacOS. Nonetheless fix will be part of 2.3.0-beta.2 which will in all likelihood be released on Monday.

**Comment by djova-dolby** [comment]
https://github.com/DolbyIO/comms-sdk-cpp/releases/tag/2.3.0-beta.2
